### PR TITLE
fix _se_to_mask

### PR DESCRIPTION
**Pull request by s-rog**
### Description
_se_to_mask currently fails with rectangular strels because the coords are specified in the wrong order

btw can't erosion and dilation share code since their relationship is just f(x) and -f(-x)?

Edit: ended up fixing erosion padding as well.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/morphology/basic_operators.py [hunk 1]
```diff
@@ -12,9 +12,9 @@ def _se_to_mask(se: torch.Tensor) -> torch.Tensor:
     num_feats = se_h * se_w
     out = torch.zeros(num_feats, 1, se_h, se_w, dtype=se.dtype, device=se.device)
     for i in range(num_feats):
-        y = i % se_h
-        x = i // se_h
-        out[i, 0, x, y] = (se_flat[i] >= 0).float()
+        y = i // se_w
+        x = i % se_w
+        out[i, 0, y, x] = (se_flat[i] >= 0).float()
```
Comment (edgarriba) [inline kornia/morphology/basic_operators.py:17]: Do you have any snippet showing this issue ? We should add it to the unit tests to verify that we fix

Comment (edgarriba) [reply kornia/morphology/basic_operators.py:17]: Could you quickly add few critical cases in the tests like this ?
https://github.com/kornia/kornia/blob/master/test/morphology/test_open.py#L19

Comment (s-rog) [reply kornia/morphology/basic_operators.py:17]: gotcha, added them to the erosion and dilation tests.

**Comment by s-rog** [comment]
hmmm weird, ran fine locally, let me look into what else I changed.

File: kornia/morphology/basic_operators.py (hunk 1 above)
Comment (edgarriba) [reply kornia/morphology/basic_operators.py:17]: does it affect to the others too ?

Comment (s-rog) [reply kornia/morphology/basic_operators.py:17]: I think every function in morphology is based on erosions and dilations?

Comment (edgarriba) [reply kornia/morphology/basic_operators.py:17]: Yes, they are

**Review by edgarriba — APPROVED** [review]

**Comment by edgarriba** [comment]
@s-rog i agree that erosion and dilation structural code can be somehow merged. Please, go ahead for a refactor. One thing that would be cool is to support batched kernels too. In addition, we could have a registry of common kernels.

**Comment by s-rog** [comment]
I've tried implementing these morphological operators in various ways but none of them seem to be practical, consuming too much vram or just computing too slowly. (eg. I tried using fourier convolutions, while it's faster than normal conv for morphology, it just consumes too much vram)

I think the only way is to write a custom cuda kernel, but that's a bit out of my wheelhouse currently...
